src/plugins/dashboard/public/application/utils/use/use_dashboard_container.tsx

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { EMPTY, Subscription, merge } from 'rxjs';
+import { catchError, distinctUntilChanged, map, mapTo, startWith, switchMap } from 'rxjs/operators';
+import deepEqual from 'fast-deep-equal';
+import { EventEmitter } from 'stream';
+import { useEffect } from 'react';
+import { opensearchFilters } from '../../../../../data/public';
+import {
+  DASHBOARD_CONTAINER_TYPE,
+  DashboardContainer,
+  DashboardContainerInput,
+  DashboardPanelState,
+} from '../../embeddable';
+import {
+  ContainerOutput,
+  ErrorEmbeddable,
+  ViewMode,
+  isErrorEmbeddable,
+} from '../../../embeddable_plugin';
+import { convertSavedDashboardPanelToPanelState } from '../../lib/embeddable_saved_object_converters';
+import { DashboardEmptyScreen, DashboardEmptyScreenProps } from '../../dashboard_empty_screen';
+import { DashboardAppStateContainer, DashboardServices, SavedDashboardPanel } from '../../../types';
+
+export const useDashboardContainer = (
+  services: DashboardServices,
+  isChromeVisible: boolean,
+  eventEmitter: EventEmitter,
+  dashboardDom: HTMLElement | null,
+  savedDashboardInstance?: any,
+  appState?: DashboardAppStateContainer
+) => {
+  const [dashboardContainer, setDashboardContainer] = useState<DashboardContainer>();
+
+  useEffect(() => {
+    const getDashboardContainer = async () => {
+      try {
+        if (savedDashboardInstance && appState) {
+          let dashboardContainerEmbeddable: DashboardContainer | undefined;
+          try {
+            dashboardContainerEmbeddable = await createDashboardEmbeddable(
+              savedDashboardInstance,
+              services,
+              appState
+            );
+          } catch (error) {
+            console.log(error);
+          }
+          setDashboardContainer(dashboardContainerEmbeddable);
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    getDashboardContainer();
```

Comment:
do you mean why do we need to call the function `getDashboardContainer()` here? 

The purpose of this function is to create a dashboard container embeddable and it will be the embeddable that can accept other visualizations as child embeddable.